### ISSUE TITLE
How to understand the codebase, how to apply my python functions into this code

### ISSUE THREAD

**Issue by duongkstn** [NVIDIA/FasterTransformer#403]
Hi, I am a python and Huggingface's `transformers` user. My model is based on T5/BART but with further generating function implementaions by myself in python (add more `LogitsProcessor` functions in [Huggingface code](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/utils.py) ). How can I use my functions along with `FasterTransformer` properly? It is really hard for me to read C++ code and understand the flow. Any advice ?

**Comment by byshiue** [comment]
There are two choices: 
1. Add your logitProcessor into c++ code, this requires to write some cuda kernel. 
2. Encapsulate the transformer block as a python op. You can refer the [GptDecoderOp](https://github.com/NVIDIA/FasterTransformer/blob/main/src/fastertransformer/th_op/gpt/GptOp.cc). This requires to understand the implementation details of FT.

**Comment by TheExGenesis** [comment]
> There are two choices:
> 
> 1. Add your logitProcessor into c++ code, this requires to write some cuda kernel.
> 2. Encapsulate the transformer block as a python op. You can refer the [GptDecoderOp](https://github.com/NVIDIA/FasterTransformer/blob/main/src/fastertransformer/th_op/gpt/GptOp.cc). This requires to understand the implementation details of FT.

Would adding a logit processor **require** writing a cuda kernel, or could it be plain C++? e.g. if it's just masking some weigh

**Comment by byshiue** [comment]
All operation in FT are on CUDA.
